vassal-app/src/main/java/VASSAL/build/module/map/boardPicker/board/mapgrid/PolygonEditor.java

```diff
@@ -373,6 +409,62 @@ else if (SwingUtils.isContextMouseButtonDown(e)) {
         insertVertex(polygon, ins, e.getX(), e.getY());
         selected = ins;
         repaint();
+
+        if (myConfigurer != null) {
+          myConfigurer.updateCoords(polygon);
+          if (selected >= 0) {
+            myConfigurer.updateCoord(polygon.xpoints[selected], polygon.ypoints[selected]);
+          }
+          else {
+            myConfigurer.updateCoord("");
+          }
+        }
+      }
+    }
+
+    @Override
+    public void keyReleased(KeyEvent e) {
+    }
+
+    @Override
+    public void keyTyped(KeyEvent e) {
+    }
+
+    @Override
+    public void keyPressed(KeyEvent e) {
+      if (selected < 0) {
+        return;
+      }
+
+      int dx = 0, dy = 0, delta = 1;
+
+      if (e.isShiftDown()) {
+        delta = 5;
+      }
+
+      switch (e.getKeyCode()) {
+      case KeyEvent.VK_UP:
+        dy = -delta;
+        break;
+      case KeyEvent.VK_DOWN:
+        dy = delta;
+        break;
+      case KeyEvent.VK_LEFT:
+        dx = -delta;
+        break;
+      case KeyEvent.VK_RIGHT:
+        dx = delta;
+        break;
+      default:
+        return;
+      }
+
+      moveVertex(polygon, selected, polygon.xpoints[selected] + dx, polygon.ypoints[selected] + dy);
+      if (myConfigurer != null) {
+        if (selected >= 0) {
```

Comment:
You know `selected >= 0` if you're here, because you return at the top of the method if `selected < 0` and don't change it in the meantime.

Reply:
Oh right I fixed this before but then probably accidentally reverted it during some debugging/experimentation